plat/common/paging.c

```diff
@@ -1418,3 +1421,101 @@ void ukplat_page_kunmap(struct uk_pagetable *pt, __vaddr_t vaddr,
 
 	pgarch_kunmap(pt, vaddr, len);
 }
+
+static inline unsigned long bootinfo_to_page_attr(__u16 flags)
+{
+	unsigned long prot = 0;
+
+	if (flags & UKPLAT_MEMRF_READ)
+		prot |= PAGE_ATTR_PROT_READ;
+	if (flags & UKPLAT_MEMRF_WRITE)
+		prot |= PAGE_ATTR_PROT_WRITE;
+	if (flags & UKPLAT_MEMRF_EXECUTE)
+		prot |= PAGE_ATTR_PROT_EXEC;
+
+	return prot;
+}
+
+int ukplat_paging_init(void)
```

Comment:
Yep, that would more sense for me as well, you are right :D.

Reply:
Yes exactly, this will be in the merging of `setup.c` logic into `ukboot`. On whether to keep it into `ukplat_paging_init()` until then, perhaps @marcrittinghaus can provide his feedback, being the author of the paging API.